device-backend/control/planktoscopehat/main.py

```
@@ -84,25 +85,34 @@ def handler_stop_signals(signum, frame):
         f"This PlanktoScope's machine name is {planktoscope.identity.load_machine_name()}"
     )
 
+    try:
+        with open("/home/pi/PlanktoScope/hardware.json", "r") as config_file:
+            configuration = json.load(config_file)
```

As mentioned in #578, this would be a great opportunity to log something (using the loguru-based `logger`) when `json.load` fails to parse the open file and throws some kind of exception (which is not a `FileNotFoundError` and thus is not caught by any exception handler which would write the exception traceback to `logger`, as far as I can tell).

Maybe one option would be to move everything currently under `if __name__ == '__main__':` into a new `main` function, and then to just have a top-level catch-all exception handler around `main()`, e.g. like the following code which might not be syntactically correct:
```
if __name__ == '__main__':
  try:
    main()
  except e:
    logger.exception('uncaught exception:', e)
    raise e
```
Adding this kind of top-level exception handler into this PR might be sufficient to mark #553 as resolved once this PR is merged.

---

On second thought, now that I've started #602 I think I'd like to make the change described above in that PR instead of this PR, so that the top-level exception handler will be truly top-level (i.e. in the `if __name__ == '__main__'` block of `~/PlanktoScope/device-backend/control/main.py`); in that case, #602 will resolve #553. Once you merge this PR, then in #602 I can move code in the `adafruithat/main.py` and `planktoscopehat/main.py` scripts into `def main()` function blocks so that those `main` functions can be invoked via module imports from `~/PlanktoScope/device-backend/control/main.py`.